src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/PermissionChecker.java

```diff
@@ -0,0 +1,94 @@
+package com.ericsson.bss.cassandra.ecaudit.auth;
+
+import java.util.Set;
+
+import org.apache.cassandra.auth.AuthenticatedUser;
+import org.apache.cassandra.auth.IAuthorizer;
+import org.apache.cassandra.auth.IResource;
+import org.apache.cassandra.auth.Permission;
+import org.apache.cassandra.auth.Resources;
+import org.apache.cassandra.auth.RoleOptions;
+import org.apache.cassandra.auth.RoleResource;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.UnauthorizedException;
+
+class PermissionChecker
+{
+    private IAuthorizer authorizer;
+
+    PermissionChecker()
+    {
+        this.authorizer = null;
+    }
+
+    PermissionChecker(IAuthorizer authorizer)
+    {
+        this.authorizer = authorizer;
+    }
+
+    private IAuthorizer getAuthorizer()
+    {
+        if (authorizer == null)
+        {
+            return getAuthorizerSync();
+        }
+
+        return authorizer;
+    }
+
+    private synchronized IAuthorizer getAuthorizerSync()
+    {
+        if (authorizer == null)
+        {
+            authorizer = DatabaseDescriptor.getAuthorizer();
+        }
+
+        return authorizer;
+    }
+
+    void checkAlterRoleAccess(AuthenticatedUser performer, RoleResource role, RoleOptions options)
```

Comment:
Structure of this method isn't very elegant. Factor out condition checks to separate methods to increase readability.